frame/staking/src/testing_utils.rs

```diff
@@ -320,29 +265,115 @@ pub fn get_seq_phragmen_solution<T: Trait>(
 	offchain_election::prepare_submission::<T>(assignments, winners, do_reduce).unwrap()
 }
 
-/// Remove all validator, nominators, votes and exposures.
-pub fn clean<T: Trait>(era: EraIndex)
-	where
-		<T as frame_system::Trait>::AccountId: codec::EncodeLike<u32>,
-		u32: codec::EncodeLike<T::AccountId>,
-{
-	<Validators<T>>::iter().for_each(|(k, _)| {
-		let ctrl = <Module<T>>::bonded(&k).unwrap();
-		<Bonded<T>>::remove(&k);
-		<Validators<T>>::remove(&k);
-		<Ledger<T>>::remove(&ctrl);
-		<ErasStakers<T>>::remove(k, era);
-	});
-	<Nominators<T>>::iter().for_each(|(k, _)| <Nominators<T>>::remove(k));
-	<Ledger<T>>::remove_all();
-	<Bonded<T>>::remove_all();
-	<QueuedElected<T>>::kill();
-	QueuedScore::kill();
+/// Returns a solution in which only one winner is elected with just a self vote.
+pub fn get_single_winner_solution<T: Trait>(
+	winner: T::AccountId
+) -> Result<(Vec<ValidatorIndex>, CompactAssignments, PhragmenScore, ElectionSize), &'static str> {
+	let snapshot_validators = <Module<T>>::snapshot_validators().unwrap();
+	let snapshot_nominators = <Module<T>>::snapshot_nominators().unwrap();
+
+	let val_index = snapshot_validators.iter().position(|x| *x == winner).ok_or("not a validator")?;
+	let nom_index = snapshot_nominators.iter().position(|x| *x == winner).ok_or("not a nominator")?;
+
+	let stake = <Staking<T>>::slashable_balance_of(&winner);
+	let stake = <T::CurrencyToVote as Convert<BalanceOf<T>, VoteWeight>>::convert(stake)
+		as ExtendedBalance;
+
+	let val_index = val_index as ValidatorIndex;
+	let nom_index = nom_index as NominatorIndex;
+
+	let winners = vec![val_index];
+	let compact = CompactAssignments {
+		votes1: vec![(nom_index, val_index)],
+		..Default::default()
+	};
+	let score = [stake, stake, stake * stake];
+	let size = ElectionSize {
+		validators: snapshot_validators.len() as ValidatorIndex,
+		nominators: snapshot_nominators.len() as NominatorIndex,
+	};
+
+	Ok((winners, compact, score, size))
 }
 
 /// get the active era.
-pub fn active_era<T: Trait>() -> EraIndex {
-	<Module<T>>::active_era().unwrap().index
+pub fn current_era<T: Trait>() -> EraIndex {
+	<Module<T>>::current_era().unwrap_or(0)
+}
+
+/// Trims a compact assignment to contain only up to the given count edges.
+pub fn trim_compact_to_assignments<T: Trait>(
```

Comment:
shouldn't we remove it now ?

Reply:
Yeah I don't think we'll ever need it again.